File: phone-exchange-2-python-examples/get_exchange_info_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../phone-exchange-2-python/SOAP"))

from get_exchange_info_soap import GetExchangeInfoSoap


def get_exchange_info_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n------------------------------------------")
    print("PhoneExchange - GetExchangeInfo - SOAP SDK")
    print("------------------------------------------")

    phone_number = "805-555-1212"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Phone Number   : {phone_number}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = GetExchangeInfoSoap(license_key, is_live, timeout_seconds * 1000)
        response = service.get_exchange_info(phone_number)

        if not hasattr(response, 'Error') or not response.Error:
            print("\r\n* Exchange Info *\r\n")
            if hasattr(response, 'ExchangeInfoResults') and response.ExchangeInfoResults:
                items = response.ExchangeInfoResults.ExchangeInfo if hasattr(response.ExchangeInfoResults, 'ExchangeInfo') else []
                if not isinstance(items, list):
                    items = [items]
                for exchange_info in items:
                    print(f"Phone Number     : {getattr(exchange_info, 'PhoneNumber', '')}")
                    print(f"Name             : {getattr(exchange_info, 'Name', '')}")
                    print(f"City             : {getattr(exchange_info, 'City', '')}")
                    print(f"State            : {getattr(exchange_info, 'State', '')}")
                    print(f"Country          : {getattr(exchange_info, 'Country', '')}")
                    print(f"Line Type        : {getattr(exchange_info, 'LineType', '')}")
                    print(f"Time Zone        : {getattr(exchange_info, 'TimeZone', '')}")
                    print(f"Latitude         : {getattr(exchange_info, 'Latitude', '')}")
                    print(f"Longitude        : {getattr(exchange_info, 'Longitude', '')}")
                    print(f"SMS Address      : {getattr(exchange_info, 'SMSAddress', '')}")
                    print(f"MMS Address      : {getattr(exchange_info, 'MMSAddress', '')}")
                    print(f"Note Codes       : {getattr(exchange_info, 'NoteCodes', [])}")
                    print(f"Note Descriptions: {getattr(exchange_info, 'NoteDescriptions', [])}")

                    print("\r\n* Ported Info *\r\n")
                    if hasattr(exchange_info, 'PortedInfo') and exchange_info.PortedInfo:
                        print(f"Original Name     : {getattr(exchange_info.PortedInfo, 'OriginalName', '')}")
                        print(f"Original Line Type: {getattr(exchange_info.PortedInfo, 'OriginalLineType', '')}")
                        print(f"Ported Date       : {getattr(exchange_info.PortedInfo, 'PortedDate', '')}")
                        print(f"LATA              : {getattr(exchange_info.PortedInfo, 'LATA', '')}")
                    else:
                        print("No ported info found.")
            else:
                print("No exchange info items found.")
        else:
            print("No exchange info found.")

        if hasattr(response, 'Error') and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Type    : {response.Error.Type}")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error TypeCode: {response.Error.TypeCode}")
            print(f"Error DescCode: {response.Error.DescCode}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Exception occurred: {str(e)}")

File: phone-exchange-2-nodejs-examples/get_exchange_info_soap_sdk_example.js
import { GetExchangeInfoSoap } from '../phone-exchange-2-nodejs/SOAP/get_exchange_info_soap.js';

async function GetExchangeInfoSoapGo(licenseKey, isLive) {
    console.log("\n-------------------------------------------");
    console.log("PhoneExchange2 - GetExchangeInfo - SOAP SDK");
    console.log("-------------------------------------------");

    const phoneNumber = "+18059631700";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Phone Number   : ${phoneNumber}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const pe2 = new GetExchangeInfoSoap(phoneNumber,licenseKey,isLive, timeoutSeconds);
        const response = await pe2.getExchangeInfoSoap();

        console.log("\n* Exchange Info *\n");
        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Type    : ${response.Error.Type}`);
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error TypeCode: ${response.Error.TypeCode}`);
            console.log(`Error DescCode: ${response.Error.DescCode}`);
        }

        if (response.ExchangeInfoResults) {
            console.log(`Phone Number     : ${response.ExchangeInfoResults.ExchangeInfo[0].PhoneNumber}`);
            console.log(`Carrier Name     : ${response.ExchangeInfoResults.ExchangeInfo[0].Name}`);
            console.log(`City             : ${response.ExchangeInfoResults.ExchangeInfo[0].City}`);
            console.log(`State            : ${response.ExchangeInfoResults.ExchangeInfo[0].State}`);
            console.log(`Country          : ${response.ExchangeInfoResults.ExchangeInfo[0].Country}`);
            console.log(`Line Type        : ${response.ExchangeInfoResults.ExchangeInfo[0].LineType}`);
            console.log(`Time Zone        : ${response.ExchangeInfoResults.ExchangeInfo[0].TimeZone}`);
            console.log(`Latitude         : ${response.ExchangeInfoResults.ExchangeInfo[0].Latitude}`);
            console.log(`Longitude        : ${response.ExchangeInfoResults.ExchangeInfo[0].Longitude}`);
            console.log(`SMS Address      : ${response.ExchangeInfoResults.ExchangeInfo[0].SMSAddress}`);
            console.log(`MMS Address      : ${response.ExchangeInfoResults.ExchangeInfo[0].MMSAddress}`);
            console.log(`Note Codes       : ${response.ExchangeInfoResults.ExchangeInfo[0].NoteCodes}`);
            console.log(`Note Descriptions: ${response.ExchangeInfoResults.ExchangeInfo[0].NoteDescriptions}`);

            console.log("\n* Ported Info *\n");
            if (response.ExchangeInfoResults.ExchangeInfo[0].PortedInfo) {
                console.log(`Original Carrier Name: ${response.ExchangeInfoResults.ExchangeInfo[0].PortedInfo.OriginalName}`);
                console.log(`Original Line Type   : ${response.ExchangeInfoResults.ExchangeInfo[0].PortedInfo.OriginalLineType}`);
                console.log(`Ported Date          : ${response.ExchangeInfoResults.ExchangeInfo[0].PortedInfo.PortedDate}`);
                console.log(`LATA                 : ${response.ExchangeInfoResults.ExchangeInfo[0].PortedInfo.LATA}`);
            } else {
                console.log("No ported info found.");
            }
        } else {
            console.log("No exchange info results found.");
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetExchangeInfoSoapGo };